frontend/src/component/ui/Table.tsx

```diff
@@ -0,0 +1,66 @@
+import React, { Component, ReactChild, ReactText } from 'react';
+
+import './Table.scss';
+
+interface IProps {
+  headings?: ReactChild[];
+  rows?: ReactChild[][];
+}
+
+export class Table extends Component<IProps> {
+  private constructHeadIfExists(headings: ReactChild[] | undefined) {
+    if (!headings || headings.length === 0) {
+      return null;
+    }
+    return <tr>{this.constructRow(headings, true)}</tr>;
+  }
+
+  private constructRow(row: ReactChild[], isHeader: boolean) {
+    return row.map((cell: ReactChild, cellIndex: number) => {
+      return isHeader
+        ? this.constructHeadingCell(cellIndex, cell)
+        : this.constructBodyCell(cellIndex, cell);
+    });
+  }
```

Comment:
Let's separate body and header construction into different functions to reduce complexity.

Reply:
This also allows us to remove the `isHeader` toggle.